### PR TITLE
PIM-10133 - PIM-10134: Prevent deletion of a channel used in a published product/shared catalog export job

### PR DESCRIPTION
**Pull request by nmarniesse**
**Description (for Contributor and Core Developer)**

- Backport of https://github.com/akeneo/pim-community-dev/pull/15472  
- prevent deletion of a channel used in a published product export

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [x] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: tests/back/Channel/Integration/Channel/Query/Sql/IsChannelUsedInProductProductExportJobIntegration.php [hunk 1]
```diff
@@ -29,13 +29,29 @@ public function test_it_determines_if_a_channel_is_used_in_product_export_jobs()
         $this->assertTrue($isChannelUsedInProductProductExportJob->execute('mobile'));
     }
 
-    private function givenAProductImportJob(): void
+    public function test_it_determines_if_a_channel_is_used_in_published_product_export_jobs(): void
```
Comment (nmarniesse) [inline tests/back/Channel/Integration/Channel/Query/Sql/IsChannelUsedInProductProductExportJobIntegration.php:32]: Maybe strange to have this test in CE as published product is a EE feature. I can move it in EE (with the service declaration), but I think it's simpler like that

Comment (mmetayer) [reply tests/back/Channel/Integration/Channel/Query/Sql/IsChannelUsedInProductProductExportJobIntegration.php:32]: no strong opinion on this. It does feel weird to have an EE job name here, but at the same time it allows us not to override the service definition in EE, thus preventing us from any mistake in the future (e.g adding a restricted job name in CE and forgetting to add it in the EE service definition override as well)

**Review by mmetayer — APPROVED** [review]

File: tests/back/Channel/Integration/Channel/Query/Sql/IsChannelUsedInProductProductExportJobIntegration.php (hunk 1 above)
Comment (ahocquard) [reply tests/back/Channel/Integration/Channel/Query/Sql/IsChannelUsedInProductProductExportJobIntegration.php:32]: The most problematic issue is the coupling it creates. It should prevent it at validation step, through a registry asking "can my channel be deleted"?

Published is coupled to channel, not the opposite. And the test would be logically inside published product.

**Comment by ahocquard** [comment]
Could we create a card for this technical debt as it creates more and more coupling? Thanks!

**Comment by nmarniesse** [comment]
> Could we create a card for this technical debt as it creates more and more coupling? Thanks!

Done https://akeneo.atlassian.net/browse/TIP-1551

**Review by LaurentPetard — APPROVED** [review]

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**